libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -4847,29 +4847,34 @@ void GCS_MAVLINK::send_generator_status() const
 
 void GCS_MAVLINK::send_water_depth() const
 {
-#if APM_BUILD_TYPE(APM_BUILD_Rover)
+#if !APM_BUILD_TYPE(APM_BUILD_Rover)
     if (!HAVE_PAYLOAD_SPACE(chan, WATER_DEPTH)) {
         return;
     }
 
     RangeFinder *rangefinder = RangeFinder::get_singleton();
-    if (rangefinder == nullptr || !rangefinder->has_data_orient(ROTATION_PITCH_270)) {
-        // no rangefinder or not facing downwards
-        return;
-    }
 
-    const bool sensor_healthy = (rangefinder->status_orient(ROTATION_PITCH_270) == RangeFinder::Status::Good);
+    for (uint8_t i=0; i<RANGEFINDER_MAX_INSTANCES; i++) {
+        AP_RangeFinder_Backend *s = rangefinder->get_backend(i);
+        
+        if (s == nullptr || s->orientation() != ROTATION_PITCH_270) {
+            continue;
+        }
 
-    // get position
-    const AP_AHRS &ahrs = AP::ahrs();
-    Location loc;
-    IGNORE_RETURN(ahrs.get_position(loc));
+        // get position
+        const AP_AHRS &ahrs = AP::ahrs();
+        Location loc;
+        IGNORE_RETURN(ahrs.get_position(loc));
+
+        // get temperature
+        float temp_C;
+        if (!s->get_temp(temp_C)) {
+            temp_C = 0.0f;
+        }
 
-    // get temperature
-    float temp_C = 0.0f;
-    IGNORE_RETURN(rangefinder->get_temp(ROTATION_PITCH_270, temp_C));
+        const bool sensor_healthy = (s->status() == RangeFinder::Status::Good);
 
-    mavlink_msg_water_depth_send(
+        mavlink_msg_water_depth_send(
```

Comment:
White space fix. And you need to insert the instance ID# here.

```suggestion
        mavlink_msg_water_depth_send(
            chan,
            AP_HAL::millis(),   // time since system boot TODO: take time of measurement
            i,                  // sensor id
            sensor_healthy,     // sensor healthy
            loc.lat,            // latitude of vehicle
            loc.lng,            // longitude of vehicle
            loc.alt * 0.01f,    // altitude of vehicle (MSL)
            ahrs.get_roll(),    // roll in radians
            ahrs.get_pitch(),   // pitch in radians
            ahrs.get_yaw(),     // yaw in radians
            s->distance_cm() * 0.01f,    // distance in meters
            temp_C);            // temperature in degC
```